pandas/tests/reshape/concat/test_index.py

```diff
@@ -323,3 +324,27 @@ def test_concat_multiindex_(self):
             {"col": ["a", "b", "c"]}, index=MultiIndex.from_product(iterables)
         )
         tm.assert_frame_equal(result_df, expected_df)
+
+    @pytest.mark.parametrize(
+        "keys",
+        [["x", "y", "x"]],
+    )
+    def test_concat_with_key_not_unique(
+        self,
+        keys: list,
+    ):
+        # GitHub #46519
+        df1 = DataFrame({"name": [1]})
+        df2 = DataFrame({"name": [2]})
+        df3 = DataFrame({"name": [3]})
+        df_a = concat([df1, df2, df3], keys=keys)
+        with tm.assert_produces_warning(PerformanceWarning):
+            out_a = df_a.loc[("x", 0), :]
+
+        df_b = DataFrame(
+            {"name": [1, 2, 3]}, index=Index([("x", 0), ("y", 0), ("x", 0)])
```

Comment:
can you also create directly as well as per this comment and compare: https://github.com/pandas-dev/pandas/issues/46519#issuecomment-1080670779